test/silhouette.jl

```diff
@@ -34,4 +34,21 @@ c = [3, 1]
 
 @test silhouettes(a, c, D) ≈ [0.5, 0.5, -1/3, 0.0]
 
+@testset "silhouette() errors when degenerated clustering is given" begin
+    a = [fill(1, 5); fill(2, 5)]
+    d = fill(0, (10, 10))
+
+    @test silhouettes(a, d) ≈ fill(0.0, 10)
+
+    d = fill(1, (10, 10))
+    d[1, 2] = d[2, 1] = 5
+    
+    @test silhouettes(a, d) ≈ [[-0.5, -0.5]; fill(0.0, 8)]
```

Comment:
`==` as well

Reply:
While knowing internal implementation of Julia type conversations of Julia it's true that `==` holds it still may not be advisable to compare floating point quantities with equality unless your scope of testing is for ensuring Julia division operator is behaving `x/y` properly for type conversions. Request you to reconsider this change suggested.

Reply:
It's fine to keep approx at L46, although I still don't understand what corner case we are testing here exactly.
But please use `==` at L41 as there we are checking the definition of `silhoutte()` when `a[i] == b[i]`. We need to make sure that it's exactly zero, not some 1e-8.